numpy/random/mtrand/mtrand.pyx

```diff
@@ -1551,18 +1551,17 @@ cdef class RandomState:
         cdef double flow, fhigh, fscale
         cdef object temp
 
-        flow = PyFloat_AsDouble(low)
-        fhigh = PyFloat_AsDouble(high)
+        if np.array(low).shape == np.array(high).shape == ():
```

Comment:
I should add that while I do not know the validity of the chained comparisons in Cython, I know it is valid in pure Python.  I guess since you can compile pure Python with Cython, it should then be valid by extension?  A quick look over the Cython docs does not indicate any issues with handling such chains.